azkaban-common/src/main/java/azkaban/utils/AzkabanPatternLayout.java

```diff
@@ -0,0 +1,112 @@
+package azkaban.utils;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import org.apache.log4j.helpers.PatternConverter;
+import org.apache.log4j.helpers.PatternParser;
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.spi.LoggingEvent;
+
+
+/**
+ * When we use the log4j Kafka appender, it seems that the appender simply does not log the stack trace anywhere
+ * Seeing as the stack trace is a very important piece of information, we create our own PatternLayout class that
+ * appends the stack trace to the log message that reported it, so that all the information regarding that error
+ * can be found one in place. In addition, we create a new PatternParser to deliver logs with current host name.
+ */
+public class AzkabanPatternLayout extends PatternLayout {
+  protected String host;
+
+  public AzkabanPatternLayout(String s) {
+    super(s);
+  }
+
+  public AzkabanPatternLayout() {
+    super();
+  }
+
+  // TODO: There is no easy way to get Jetty Server's port by InetAddress API.
+  // We might need to find some other way to find this application's port
+  protected String getPort() {
+    return "";
```

Comment:
The Azkaban Server object has methods for obtaining the port information. However, we need to check if accessing that class directly is the best option.